### PR TITLE
feat(handlers): Calls for incito section visible and hidden tracking events, offer view click and long offer view click

### PR DESCRIPTION
**Pull request by klarstrup**
Including a refactor of handler calling logic into a simpler `emit` interface that tries to call the handler of a given name for both platforms

### CODE REVIEW

**Review by mortenbo — COMMENTED** [review]
One thing I don't like is that some events are prefixed with "incito" while others aren't. What's the idea behind that?

**Comment by lozhuf** [comment]
`incitoFinishedRendering` is an old event that is already being used, so we cant change the name, but the `incitoSection*` ones are new... dunno if or when it makes sense to include the incito word in the event name. I guess it's kind if implied that it's an incito section by the fact this is an incito web view ;P

**Comment by klarstrup** [comment]
I don't feel strongly about the naming, @lozhuf @AgneseBussone can you confirm that this updated html doesn't break existing implementations? https://raw.githubusercontent.com/tjek/incito-webview/5f9df68ed577fdb393d358497ec6115aded1c09b/index-1.0.0.html

**Comment by AgneseBussone** [comment]
Why not releasing a `.1` version, so you could cleanup all the names and add new callbacks freely?

**Comment by klarstrup** [comment]
> Why not releasing a `.1` version, so you could cleanup all the names and add new callbacks freely?

It'd be a `2.0` version if we break compatibility, which is fine by me but might compound maintenance work in the future as every version will need any relevant fixes applied retroactively.

**Comment by AgneseBussone** [comment]
@klarstrup I seem to get this error from the webview console using your branch
```
Uncaught Error: Java bridge method can't be invoked on a non-injected object -- From line 54 of file:///android_asset/incito/webview/index-1.0.0.html
```
and the incito doesn't load.

File: index-1.0.0.html [hunk 1]
```diff
@@ -40,6 +40,34 @@
...
   <body>
     <script>
+      function emit(handler) {
+        var parameters = Array.prototype.slice.call(arguments, 1);
+
+        var iosHandler =
+          window.webkit &&
+          window.webkit.messageHandlers &&
+          window.webkit.messageHandlers[handler];
+        if (iosHandler) iosHandler.postMessage.apply(null, parameters);
```
Comment (lozhuf) [inline index-1.0.0.html]: This doesnt work on iOS. Changing it to this works:
```
if (iosHandler) iosHandler.postMessage(parameters);
```

**Comment by klarstrup** [comment]
I've made changes such that the handlers are called directly, it's annoying that `progress` has two arguments since i think this only works as a unary handler on iOS but please check out the newest version

**Comment by AgneseBussone** [comment]
@klarstrup 
I still have an error and it doesn't load the incito
`Uncaught Error: Method not found -- From line 57 of file:///android_asset/incito/webview/index-1.0.0.html`

But it seems that the very first section appeared event triggers the callback.

**Comment by klarstrup** [comment]
Is that the entire error? :/ No stack trace?

**Comment by AgneseBussone** [comment]
No, that's all I see. 
I setup the webview logs like this
```
// Catch javascript console messages
        incitoWebView?.webChromeClient = object : WebChromeClient() {
            override fun onConsoleMessage(cm: ConsoleMessage): Boolean {
                SgnLog.d(
                    TAG,
                    "Webview console message\n${cm.message()} -- From line ${cm.lineNumber()} of ${cm.sourceId()}"
                )
                return true
            }
        }
```
as described [here](https://developer.android.com/develop/ui/views/layout/webapps/debugging#WebView)

**Review by AgneseBussone — APPROVED** [review]

**Comment by klarstrup** [comment]
How does the publish flow work from here?

**Comment by klarstrup** [comment]
> How does the publish flow work from here?

@mortenbo?

**Comment by lozhuf** [comment]
we can delete this now, right?

**Comment by klarstrup** [comment]
Ye

**Pull request merged**